tool/tsh/tsh.go

```diff
@@ -4729,3 +4735,51 @@ func onHeadlessApprove(cf *CLIConf) error {
 	})
 	return trace.Wrap(err)
 }
+
+var mlockModes = []string{mlockModeNo, mlockModeAuto, mlockModeBestEffort, mlockModeStrict}
+
+const (
+	// mlockModeNo disables locking process memory.
+	mlockModeNo = "off"
+	// mlockModeAuto automatically chooses whether memory locking will be attempted and/or enforced.
+	mlockModeAuto = "auto"
+	// mlockBestEfforts enables locking process memory, but errors will be ignored and logged.
+	mlockModeBestEffort = "best_effort"
+	// mlockModeStrict enables locking process memory and enforces it succeeds without errors.
+	mlockModeStrict = "strict"
+
+	// mlockFailureMessage is a user readable message for mlock errors and debug logs.
+	mlockFailureMessage = "Failed to lock process memory for headless login." +
+		"Memory locking is used to prevent secrets in memory from being swapped to disk. " +
+		"Please ensure that memory locking is available on your system and your user has " +
+		"locking privileges. This usually means using a Linux operating system and increasing " +
+		`your user's memory locking limit to unlimited. Please enable mlock on your system` +
+		"or set --mlock=off to disable it (not recommended in production environments)\n" +
+		"https://goteleport.com/docs/access-controls/guides/headless/#troubleshooting"
```

Comment:
The error message looks much better, a couple of notes.

> This usually means using a Linux operating system

The documentation takes a much clearer stance here ("only supported on Linux"). Can you update the language here to match the documentation?

> not recommended in production environments

Can you explain why it's not recommended in production environment here as well? Same as before the documentation is great, can we copy it over? "This is not recommended in production environments on shared systems where a memory swap attack is possible."